core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -498,6 +498,67 @@ def fit(self, **kwargs):
             logger.warning(f'\tWarning: Model has no time left to train, skipping model... (Time Left = {round(kwargs["time_limit"], 1)}s)')
             raise TimeLimitExceeded
 
+    def fit_with_prune(self, max_num_fit=3, prune_threshold=0., num_shuffle_sets=50, **kwargs):
+        """
+        Functionally identical to `fit` method, but repeats feature importance based pruning until
+        validation set performance degrades or `max_num_fit` iterations have passed.
+
+        Parameters
+        ----------
+        max_num_fit : int, default = 3
+            Maximum number of time feature selection and fitting are performed
+        prune_threshold : float, default = 0.
+            Feature importance threshold that features must meet in order to not be dropped
+        num_shuffle_sets : int, default = 3
+            Number of shuffles to evaluate when computing permutation feature importance
+        """
+        if kwargs['X_val'] is None or kwargs['y_val'] is None:
+            logger.error(f'fit_with_prune must be called with validation data.')
+            raise Exception
+
+        untrained_model_path = os.path.join(self.path, 'untrained_')
+        best_model_path = os.path.join(self.path, 'best_')
+        best_score = -float("inf")
```

Comment:
In my personal experience, better to have this be None and have special logic if it is None in the loop. Otherwise can lead to strange bugs or weird logging messages.